src/main/kotlin/com/babcock/http/HttpParser.kt

```diff
@@ -0,0 +1,169 @@
+                        val temp = dataBufferProcess.toString().trim()
+                        println("Key:$temp")
+                        name = temp
+
+                        dataBufferProcess.delete(0, dataBufferProcess.length)
+                    }
+
+                    0x0d -> {
+                        val temp = dataBufferProcess.toString().trim()
+                        println("VALUE:$temp")
+                        content = temp
+                        dataBufferProcess.delete(0, dataBufferProcess.length)
+                    }
+
+                    (0x0a) -> {
+                        byte = reader.read()
+                        if (byte == 0x0d) {
+                            println("End")
+                            break
+                        } else {
+                            dataBufferProcess.append(byte.toChar())
+                        }
+                    }
+
+                    else -> {
+                        dataBufferProcess.append(byte.toChar())
+                        request.headers[name] = content
+                    }
+
+                }
+            }
+        }
+
+        println(dataBufferProcess.toString())
+        for (i in request.headers) {
+            println(i)
+        }
+
+
+    }
+
+
+    @Throws(HttpParseException::class)
+    private fun parseRequestLine(reader: InputStreamReader, request: HttpReq): HttpReq {
+        var byte: Int
+        val dataBuffer = StringBuilder()
+
+
+        var parsedMethod: Boolean = false
+        var parsedRequestTarget: Boolean = false
+
+        while ((reader.read().also { byte = it }) >= 0) {
+            if (byte == CR) {
+                byte = reader.read()
+                if (byte == LF) {
+                    log.logWarning("Request line VERSION to process: $dataBuffer")
+
+                    if (!parsedMethod || !parsedRequestTarget) {
+                        request.statusCode = HttpStatusCode.CLIENT_ERROR_400_BAD_REQUEST
+                        request.statusNumber = HttpStatusCode.CLIENT_ERROR_400_BAD_REQUEST.STATUS_CODE.toString()
+                        request.statusMsg = HttpStatusCode.CLIENT_ERROR_400_BAD_REQUEST.MESSAGE
+                        return request
+                        //throw HttpParseException(HttpStatusCode.CLIENT_ERROR_400_BAD_REQUEST)
+                    }
+                    try {
+                        log.logSuccess("Version test: ${dataBuffer.toString()}")
+                        request.compatibleHttpVersion = dataBuffer.toString()
+                    } catch (e: HttpParseException) {
+                        request.statusCode = HttpStatusCode.CLIENT_ERROR_400_BAD_REQUEST
+                        request.statusNumber = HttpStatusCode.CLIENT_ERROR_400_BAD_REQUEST.STATUS_CODE.toString()
+                        request.statusMsg = HttpStatusCode.CLIENT_ERROR_400_BAD_REQUEST.MESSAGE
+                        //throw HttpParseException(HttpStatusCode.CLIENT_ERROR_400_BAD_REQUEST)
```

Comment:
I see a lot of these commented `throw SomeKindOfError` – do you need them? Some won't be reached as they're after returns

Reply:
Initially I thought I wanted to throw exceptions as a way to debug and test. I didn't have a fleshed out response object at the time to know what kind of code I was getting when I had a bad request. It seemed logical at the time but now I think I will scrap this.